modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/ResolverCache.scala

```diff
@@ -118,21 +156,31 @@ object ResolverCache {
       Applicative[F].pure(none)
   }
 
-  def getItem[F[_]: Monad: Clock, A, K](
+  private[resolver] def shouldCreateResolverCache(size: Int, ttl: Option[TTL]): Boolean =
+    size > 0 && ttl.forall(_ > 0.seconds)
+
+  private def getTimestampedItem[F[_]: Monad: Clock, A, K](
     ttl: Option[TTL],
     c: LruMap[F, K, CacheEntry[Lookup[A]]],
     key: K
-  ): F[Option[Lookup[A]]] = {
+  ): F[Option[TimestampedItem[Lookup[A]]]] = {
     val wrapped = for {
-      (storageTime, lookup) <- OptionT(c.get(key))
-      currentTime           <- OptionT.liftF(Clock[F].realTime)
+      (storageTime: StorageTime, lookup) <- OptionT(c.get(key))
+      currentTime                        <- OptionT.liftF(Clock[F].realTime)
       _ = currentTime
       if isViable(ttl, currentTime, storageTime) || lookup.isLeft // isLeft
-    } yield lookup
+    } yield TimestampedItem(lookup, storageTime)
 
     wrapped.value
   }
 
+  def getItem[F[_]: Monad: Clock, A, K](
+    ttl: Option[TTL],
+    c: LruMap[F, K, (TTL, Lookup[A])],
```

Comment:
That `TTL` should be... `StorageTime`? It looks like it was `TTL` before your changes, but I think it's not correct...? Can we change it? OR can we use `CacheEntry[Lookup[A]]` for this tuple (and other tuples in this file)? Whatever you think is the easiest to read :)

Reply:
Good point. I like `CacheEntry[Lookup[A]]`. Have fixed it now.